src/connectors/hits/connectHits.js

```diff
@@ -82,12 +82,8 @@ export default function connectHits(renderFn, unmountFn) {
       },
 
       render({ results, instantSearchInstance }) {
-        if (
-          widgetParams.escapeHTML &&
-          results.hits &&
-          results.hits.length > 0
-        ) {
-          results.hits = escapeHTML(results.hits);
+        if (escapeHTML && results.hits && results.hits.length > 0) {
+          results.hits = escapeHits(results.hits);
```

Comment:
I remember a bug when trying to SSR in Vue due to this rewriting. @samouss, do you remember the context? Maybe we can prevent the modification here?

Reply:
The issue was on our side because we did not have the same configuration on the server and the client. But it happen because we cache the escaped hits directly on the results (mutate the results actually). 

We can solve the issue later to force the usage of the highlight helper. We won't have to escape the hits inside the connector but only at the highlight helper level. Note that we should still have the `escapeHTML` option to provide the placeholder tag.